### PR TITLE
Exports only the clients and products relevant to the requested period

### PR DESCRIPTION
**Pull request by froque**
Suppliers is no longer exported, since MovementOfGoods section is not
being exported at the moment.

### CODE REVIEW

File: billy-portugal-ebean/src/main/java/com/premiumminds/billy/portugal/services/export/saftpt/v1_02_01/PTSAFTFileGenerator.java [hunk 1]
```diff
@@ -232,35 +235,57 @@ public AuditFile runTransaction() throws Exception {
...
+
+                    List<PTReceiptInvoiceEntity> receiptInvoices = daoPTReceiptInvoice
+                            .getBusinessReceiptInvoicesForSAFTPT(businessEntity.getUID(), fromDate, toDate);
+
+                    List<PTCreditNoteEntity> creditNotes = daoPTCreditNote
+                            .getBusinessCreditNotesForSAFTPT(businessEntity.getUID(), fromDate, toDate);
+
+                    Map<Long, PTCustomerEntity> customers = new HashMap<>();
+                    Map<Long, PTProductEntity> products = new HashMap<>();
+                    for (PTGenericInvoiceEntity invoice : invoices) {
```
Comment (trcdl) [inline billy-portugal-ebean/src/main/java/com/premiumminds/billy/portugal/services/export/saftpt/v1_02_01/PTSAFTFileGenerator.java:253]: Penso que se poderia extrair-se este grupo para uma função para evitar a replicação.

File: billy-portugal-ebean/src/main/java/com/premiumminds/billy/portugal/services/export/saftpt/v1_03_01/PTSAFTFileGenerator.java [hunk 2]
```diff
@@ -235,34 +238,57 @@ public AuditFile runTransaction() throws Exception {
...
+
+                    List<PTReceiptInvoiceEntity> receiptInvoices = daoPTReceiptInvoice
+                            .getBusinessReceiptInvoicesForSAFTPT(businessEntity.getUID(), fromDate, toDate);
+
+                    List<PTCreditNoteEntity> creditNotes = daoPTCreditNote
+                            .getBusinessCreditNotesForSAFTPT(businessEntity.getUID(), fromDate, toDate);
+
+                    Map<Long, PTCustomerEntity> customers = new HashMap<>();
+                    Map<Long, PTProductEntity> products = new HashMap<>();
+                    for (PTGenericInvoiceEntity invoice : invoices) {
```
Comment (trcdl) [inline billy-portugal-ebean/src/main/java/com/premiumminds/billy/portugal/services/export/saftpt/v1_03_01/PTSAFTFileGenerator.java:256]: Same

File: billy-portugal-ebean/src/main/java/com/premiumminds/billy/portugal/services/export/saftpt/v1_04_01/PTSAFTFileGenerator.java [hunk 3]
```diff
@@ -266,34 +269,57 @@ public AuditFile runTransaction() throws Exception {
...
+
+					List<PTReceiptInvoiceEntity> receiptInvoices = daoPTReceiptInvoice
+							.getBusinessReceiptInvoicesForSAFTPT(businessEntity.getUID(), fromDate, toDate);
+
+					List<PTCreditNoteEntity> creditNotes = daoPTCreditNote
+							.getBusinessCreditNotesForSAFTPT(businessEntity.getUID(), fromDate, toDate);
+
+					Map<Long, PTCustomerEntity> customers = new HashMap<>();
+					Map<Long, PTProductEntity> products = new HashMap<>();
+					for (PTGenericInvoiceEntity invoice : invoices) {
```
Comment (trcdl) [inline billy-portugal-ebean/src/main/java/com/premiumminds/billy/portugal/services/export/saftpt/v1_04_01/PTSAFTFileGenerator.java:287]: Same.

File: billy-portugal/src/main/java/com/premiumminds/billy/portugal/services/export/saftpt/v1_02_01/PTSAFTFileGenerator.java [hunk 4]
```diff
@@ -246,34 +249,57 @@ public AuditFile runTransaction() throws Exception {
...
+
+                    List<PTReceiptInvoiceEntity> receiptInvoices = daoPTReceiptInvoice
+                            .getBusinessReceiptInvoicesForSAFTPT(businessEntity.getUID(), fromDate, toDate);
+
+                    List<PTCreditNoteEntity> creditNotes = daoPTCreditNote
+                            .getBusinessCreditNotesForSAFTPT(businessEntity.getUID(), fromDate, toDate);
+
+                    Map<Long, PTCustomerEntity> customers = new HashMap<>();
+                    Map<Long, PTProductEntity> products = new HashMap<>();
+                    for (PTGenericInvoiceEntity invoice : invoices) {
```
Comment (trcdl) [inline billy-portugal/src/main/java/com/premiumminds/billy/portugal/services/export/saftpt/v1_02_01/PTSAFTFileGenerator.java:267]: Same

File: billy-portugal/src/main/java/com/premiumminds/billy/portugal/services/export/saftpt/v1_03_01/PTSAFTFileGenerator.java [hunk 5]
```diff
@@ -248,34 +251,57 @@ public AuditFile runTransaction() throws Exception {
...
+
+                    List<PTReceiptInvoiceEntity> receiptInvoices = daoPTReceiptInvoice
+                            .getBusinessReceiptInvoicesForSAFTPT(businessEntity.getUID(), fromDate, toDate);
+
+                    List<PTCreditNoteEntity> creditNotes = daoPTCreditNote
+                            .getBusinessCreditNotesForSAFTPT(businessEntity.getUID(), fromDate, toDate);
+
+                    Map<Long, PTCustomerEntity> customers = new HashMap<>();
+                    Map<Long, PTProductEntity> products = new HashMap<>();
+                    for (PTGenericInvoiceEntity invoice : invoices) {
```
Comment (trcdl) [inline billy-portugal/src/main/java/com/premiumminds/billy/portugal/services/export/saftpt/v1_03_01/PTSAFTFileGenerator.java:269]: Same.

File: billy-portugal/src/main/java/com/premiumminds/billy/portugal/services/export/saftpt/v1_04_01/PTSAFTFileGenerator.java [hunk 6]
```diff
@@ -266,39 +269,62 @@ public AuditFile runTransaction() throws Exception {
...
+					List<PTReceiptInvoiceEntity> receiptInvoices = daoPTReceiptInvoice
+							.getBusinessReceiptInvoicesForSAFTPT(businessEntity.getUID(), fromDate, toDate);
+
+					List<PTCreditNoteEntity> creditNotes = daoPTCreditNote
+							.getBusinessCreditNotesForSAFTPT(businessEntity.getUID(), fromDate, toDate);
+
+
+					Map<Long, PTCustomerEntity> customers = new HashMap<>();
+					Map<Long, PTProductEntity> products = new HashMap<>();
+					for (PTGenericInvoiceEntity invoice : invoices) {
```
Comment (trcdl) [inline billy-portugal/src/main/java/com/premiumminds/billy/portugal/services/export/saftpt/v1_04_01/PTSAFTFileGenerator.java:288]: Same.

**Review by trcdl — APPROVED** [review]
The requested changed don't block the approval

**Comment by froque** [comment]
rebased, due to conflicts

**Comment by froque** [comment]
> The requested changed don't block the approval

maybe another time. I'm going to merge as is.

**Pull request merged**